doc/voq/voq_hld.md

```diff
@@ -0,0 +1,879 @@
+      "Ethernet3": {},
+      "Ethernet1"|"10.0.0.1/16": {},
+      "Ethernet2"|"20.0.0.1/16": {},
+      "Ethernet3"|"30.0.0.1/16": {}
+   }
+}
+```
+
+##### In ASIC# 1
+
+```
+{
+   "INTERFACE": {
+      "Ethernet128": {},
+      "Ethernet129": {},
+      "Ethernet130": {},
+      "Ethernet128"|"10.1.0.1/16": {},
+      "Ethernet129"|"20.1.0.1/16": {},
+      "Ethernet130"|"30.1.0.1/16": {}
+   }
+}
+```
+### Interface infomation for local and remote system ports in APP DB
+IP address configuration for remote system ports are done in local config DB as shown above
+
+The interface information for remote system ports are populated by VoqDbSyncd
+
+#### APP DB Objects:
+
+##### In ASIC #0
+
+```
+{
+   "INTF_TABLE": {
+      "Ethernet1": {},
+      "Ethernet2": {},
+      "Ethernet3": {},
+      "Ethernet1":"10.0.0.1/24": {},
+      "Ethernet2":"20.0.0.1/24": {},
+      "Ethernet3":"30.0.0.1/24": {},
+      "Ethernet128": {},
+      "Ethernet129": {},
+      "Ethernet130": {}
+   }
+}
+```
+##### In ASIC #1
+
+```
+{
+   "INTF_TABLE": {
+      "Ethernet1": {},
+      "Ethernet2": {},
+      "Ethernet3": {},
+       Ethernet128": {},
+      "Ethernet129": {},
+      "Ethernet130": {},
+      "Ethernet128":"10.1.0.1/24": {},
+      "Ethernet129":"20.1.0.1/24": {},
+      "Ethernet130":"30.1.0.1/24": {}
+   }
+}
+```
+
+### Static Neighbor configuration in Config DB
+This is optional. 
+
+#### Config DB objects:
+Static neighbors are configured on local ports
+
+##### On ASIC #0
+
+```
+{
+   "NEIGH": {
+      "Ethernet1:10.0.0.2": {
+         "neigh": "02:06:0a:00:00:01",
+         "vrf": "1"
+      }
+    }
+}
+```
+##### On ASIC #1
+
+```
+{
+   "NEIGH": {
+      "Ethernet128:10.1.0.2": {
+         "neigh": "02:06:1a:00:00:01",
+         "vrf": "1"
+      }
+    }
+}
+```
+### Neighbor information in APP DB
+
+#### APP DB Objects:
+The NEIGH_TABLE has entries for locally learned neighbors on local ports
+
+The NEIGH_TABLE also has entries for neighbors learned/configured on remote system ports (populated by VoqDbSyncd)
+
+##### On ASIC #0
+```
+{
+  "NEIGH_TABLE": {
+      "Ethernet1:10.0.0.2": {
+         "neigh": "02:06:0a:00:00:01",
+         "family", "IPv4",
+         "vrf": "1"
+      },
+      "Ethernet2:20.0.0.2": {
+         "neigh": "02:06:0b:00:00:01",
+         "family", "IPv4",
+         "vrf": "1"
+      },
+      "Ethernet128:10.1.0.1": {
+         "neigh": "02:16:0a:00:00:01",
+         "family": "IPv4",
+         "encap_index": "4097",
+         "vrf": "1",
+      }
+  }
+}
+```
+
+##### On ASIC #1
+```
+{
+  "NEIGH_TABLE": {
+      "Ethernet1:10.0.0.2": {
+         "neigh": "02:06:0a:00:00:01",
+         "family": "IPv4"
+         "encap_index": "4096",
+         "vrf": "1"
+      },
+      "Ethernet128:10.1.0.2": {
+         "neigh": "02:16:0a:00:00:01",
+         "vrf": "1"
+      },
+      "Ethernet129:20.1.0.2": {
+         "neigh": "0b2:16:0b:00:00:01",
+         "vrf": "1"
+      }
+    }
+}
+```
+
+# 5 References
```

Comment:
The Future work section in the Arch doc can also be migrated here because it links with the SYSTEM_PORT schema etc.

Reply:
Here is a blurb that can be incorporated:
# 4 Future Work

## 4.1 Dynamic System Ports

Dynamic system port support is required to support the following forwarding scenarios

* Addition of a new forwarding device into an existing running system
* Replacing a forwarding device with another device of a different hardware SKU (such as replacing a linecard with a new linecard of a different SKU in a chassis slot).

Both these scenarios can be supported smoothly as long as the global system port numbering scheme is maintained and the modifications to system ports can be performed without impacting the System Port IDs of the running sys\
tem.

Support for dynamic system ports requires SAI support for the `create_port` and `remove_port` calls.

### Overview of PortsOrch changes

 * Subscribe to the SYSTEM_PORT table in PortsOrch
 * React to changes in the SYSTEM_PORT table
 * Make the appropriate `create_port` and `remove_port` calls.

In addition, forwarding features that are dependent on System Ports need to react to these changes and reprogram the related forwarding plane state such as routing nexthops, LAG membership etc.

Reply:
Agreed. Will make the changes in the update to PR-639